cmd/worker/consul.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 Backup Operator Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	backupv1alpha1 "github.com/kubism/backup-operator/api/v1alpha1"
+	"github.com/kubism/backup-operator/pkg/consul"
+	"github.com/kubism/backup-operator/pkg/s3"
+	"github.com/spf13/cobra"
+)
+
+var consulCmd = &cobra.Command{
+	Use:   "consul [flags] config",
+	Short: "Backups consul using specified config",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return fmt.Errorf("config path expected as one and only argument")
+		}
+		file, err := os.Open(args[0])
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		raw, err := ioutil.ReadAll(file)
+		if err != nil {
+			return err
+		}
+		var plan backupv1alpha1.ConsulBackupPlan
+		err = json.Unmarshal([]byte(os.ExpandEnv(string(raw))), &plan)
+		if err != nil {
+			return err
+		}
+
+		username := ""
+		if v := os.Getenv("CONSUL_HTTP_USERNAME"); v != "" {
```

Comment:
This is debatable, but I would suggest keeping everything in the CRD, so that each worker is not depending on environment variables, but rather those are substituted into the mounted json.

This has the advantage that the CRD is self-documenting. However this is probably something left to personal taste and debatable.

Reply:
I think this is based on the misunderstanding above, so I would agree with your solution having it self-documented.